source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -249,7 +251,9 @@ void EnvoyQuicClientConnection::onFileEvent(uint32_t events,
   if (connected() && (events & Event::FileReadyType::Read)) {
     Api::IoErrorPtr err = Network::Utility::readPacketsFromSocket(
         connection_socket.ioHandle(), *connection_socket.connectionInfoProvider().localAddress(),
-        *this, dispatcher_.timeSource(), /*prefer_gro=*/false, packets_dropped_);
+        *this, dispatcher_.timeSource(), prefer_gro_,
+        // For client connections, recvmmsg consumes more CPU with no performance improvement.
+        /*allow_mmsg=*/false, packets_dropped_);
```

Comment:
Should allow_mmsg also be guarded by prefer_gro_?

Reply:
I added a new runtime guard for allow_mmsg, and added tests.  PTAL